README.md

```diff
@@ -277,6 +277,69 @@ end
 #=> "error message"
 ```
 
+### [Block](https://hexdocs.pm/exceptional/Exceptional.Block.html)
+
+Kind of a combination of Elixir's normal
+[`with`](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#with/1)
+special form in addition to a monad-style `do` pipeline.
+
+This automatically-wraps every return value with
+[`normalize`](https://hexdocs.pm/exceptional/Exceptional.Normalize.html).
+
+```elixir
+block do
+  a <- {:ok, 2}
+  b = a * 2
+  c <- {:ok, b * 2}
+  c * 2
+end
+#=> 16
+
+block do
+  a <- {:ok, 2}
+  b = a * 2
+  _ = 42
+  c <- {:error, "Failed: #{b}"}
+  c * 2
+end
+#=> %ErlangError{original: "Failed: 4"}
+
+conversion_fun = fn
+  {:blah, reason} -> %ErlangError{original: "Blah: #{reason}"}
+  e -> e
+end
+block conversion_fun: conversion_fun do
+  a <- {:ok, 2}
+  b = a * 2
+  _ = 42
+  c <- {:blah, "Failed: #{b}"}
+  c * 2
+else
```

Comment:
🤔 Hmm, I wonder if there's a more descriptive word than `else` here? We have something preeeeetty close to `try/rescue` here... perhaps `rescue` or `errors do`?

Reply:
Well we can only use an Elixir block keyword if we want it block level here, which I really think we should for consistency with the rest of the language, so `rescue` is possible, however I was thinking of leaving `rescue` and `catch` open for possible exception handling later too, so that really only left things like `else` and `after` and so forth, and as with `rescue`/`catch`, `after` also has possible later use with message handling (I've envisioned possible `receive` block enhancements in this as well later).  So if you have any ideas?